### PR TITLE
Buttons with Icons done. Ready for review. Urgent needed feature

### PR DESCRIPTION
**Pull request by LucaMele**
Fixes # (issue).

Changes proposed in this pull request:

Button with icon on the right only: https://design.axa.com/web-guidelines/buttons-and-links#types-of-buttons

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
 
 # How Has This Been Tested?
 
Local and Browserstack
 
 # Checklist:
 
 - [x] My code follows the style guidelines of this project
 - [x] I have performed a self-review of my own code
 - [x] I have commented my code, particularly in hard-to-understand areas
 - [x] I have made corresponding changes to the documentation
 - [x] My changes generate no new warnings
 - [x] I have added tests that prove my fix is effective or that my feature works
 - [x] New and existing unit tests pass locally with my changes
 - [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by LucaMele** [comment]
#fixes #404

**Comment by LucaMele** [comment]
DIFFERENCE TO GUIDLINE: the guideline (https://design.axa.com/web-guidelines/buttons-and-links#types-of-buttons) has NO BORDER top down on the button and therefore the separator cuts in 2 the button. Our styleguide apparently has a border top down and so it goes around the separator. I suggest lets leave it like it and gets anjas FB once she sees it and if border have to be removed or keep border and cut the button in two we can use box shadow or after and before element instead of absolut positioning + border. Is my suggestion only anyway.

**Comment by AndyOGo** [comment]
I just run it locally. Now I see what you mean, this 1 or 2 pixel gab above and below.
Yes I agree we should try an absolute positioned pseudo element instead.

Btw. the icon isn't working for **not recommended** multiline buttons, but @anja225 should decide about this.

screen shot 2018-05-01 at 11 03 07

File: src/components/m-button/_template.js [hunk 1]
```diff
@@ -5,13 +5,14 @@ import classnames from 'classnames';
 export default function ({
   tag = 'button',
   color,
-  url = '#',
+  href = '#',
```
Comment (AndyOGo) [inline src/components/m-button/_template.js]: awesome

File: src/components/m-button/_template.js [hunk 2]
```diff
@@ -20,19 +21,26 @@ export default function ({
     'm-button--motion': motion,
     'm-button--gpu': gpu,
     'm-button--arrow': arrow,
+    'm-button--generic-icon': icon && !arrow,
   });
 
-  const arrowIcon = raw('<axa-icon icon="arrow" classes="m-button__arrow"></axa-icon>');
+  let arrowIcon;
+  let genericIcon;
+  if (arrow) {
```
Comment (AndyOGo) [inline src/components/m-button/_template.js]: Great spot, much better 👍

File: src/components/m-button/mixins/_button.scss [hunk 3]
```diff
@@ -20,7 +20,7 @@
 
   position: relative;
 
-  padding: 10px $x-padding;
+  padding: $button-padding-top-bottom-md $x-padding;
```
Comment (AndyOGo) [inline src/components/m-button/mixins/_button.scss]: nice

File: src/components/m-button/mixins/_button.scss [hunk 4]
```diff
@@ -113,32 +113,68 @@
...
+
+  padding-right: 0;
+}
+
 @mixin button--sm() {
   @include typo-legals--semibold();
 
-  padding-top: 5.5px;
-  padding-bottom: 5.5px;
+  padding-top: $button-padding-top-bottom-sm;
```
Comment (AndyOGo) [inline src/components/m-button/mixins/_button.scss]: 👍

**Comment by LucaMele** [comment]
so u think we should go with the before/after element without anjas FB or merge and wait for her?

i will try to find a quick solution for the multine thing, but wont invest to much as is really bad looking if we have multilines + icons

**Comment by LucaMele** [comment]
I dont thinks is good to absolut position the icon. easier have no border and 2 after elements instead for border-top and bottom replace

File: src/components/m-button/mixins/_button.scss [hunk 5]
```diff
@@ -135,16 +129,35 @@
...
+
+  top: 0;
+  right: 0;
 
-  margin-top: - $x-padding / 2;
-  margin-bottom: - $x-padding / 2;
+  margin-top: - $button-border-width;
+  margin-bottom: 0;
+
+  height: calc(100% + #{$button-border-width});
```
Comment (AndyOGo) [inline src/components/m-button/mixins/_button.scss]: I thought if you omit `height` and use `top: -$button-border-width;` and `bottom: -$bottom-border-width` it should stretch by itself

**Pull request merged**